packages/edit-site/src/components/editor/index.js

```diff
@@ -152,12 +152,36 @@ export default function Editor() {
 	// action in <URlQueryController> from double-announcing.
 	useTitle( hasLoadedPost && title );
 
-	if ( ! hasLoadedPost ) {
-		return <CanvasSpinner />;
-	}
+	const { hasResolvingSelectors } = useSelect( ( select ) => {
+		return {
+			hasResolvingSelectors: select( coreStore ).hasResolvingSelectors(),
+		};
+	} );
+	const [ loaded, setLoaded ] = useState( false );
+	const timeoutRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! hasResolvingSelectors && ! loaded ) {
+			clearTimeout( timeoutRef.current );
+			timeoutRef.current = setTimeout( () => {
+				setLoaded( true );
+			}, 1000 );
```

Comment:
Curious to know what this arbitrary timeout is for?

Reply:
There are brief moments where there aren't any resolving selectors, but a few milliseconds later, there are resolving selectors present. This arbitrary timeout is there to ensure that we're marking the editor as loaded only if there haven't been any resolving selectors for an entire second.

Reply:
This reads a bit like an anti-pattern to me and a bad UX for users. Now every user has to wait at least a second to load the editor, no matter how fast their device is. Could there also be cases when the user's device is so slow that the 1-second delay isn't sufficient to run all the remaining tasks?

Could we figure out the correct timing to subscribe to the load event under the hood? This might be a bit more difficult to do but I think it'll be worth it 😆. At the very least, let's add some comments to explain the timeout.

Reply:
In terms of blocks, it seems like it's difficult to know what's going to load next. The next resource/block to load is revealed only once the previous resource/block has been loaded, parsed and rendered. My assumption is that's what this timeout is for.

The two ideas I had:
- Be able to parse block content to determine what resources will load next. This would require a consistent way to declare resources for blocks. Only hide the loading spinner when there are no more known resources and all resolution has finished.
- Somehow wait until all known inner blocks have rendered.

Reply:
> This reads a bit like an anti-pattern to me and a bad UX for users. Now every user has to wait at least a second to load the editor, no matter how fast their device is. Could there also be cases when the user's device is so slow that the 1-second delay isn't sufficient to run all the remaining tasks?

I don't claim for the 1s to be perfect. I claim it to be much better than what we had before, and much better than if we didn't have it all (in which case we'd be seeing bad flickering).

Yes, 1s might be insufficient for very slow devices, but we're making a compromise here and I'm happy to experiment with timing separately. Regardless, I think this is much better than what we had before, so it's worth going with it for the time being while experimenting with better timing.

> Could we figure out the correct timing to subscribe to the load event under the hood? This might be a bit more difficult to do but I think it'll be worth it 😆. At the very least, let's add some comments to explain the timeout.

I'm open to suggestions. From my understanding, there isn't a good way to predict or subscribe to that, nor to figure out the perfect timing without a lot of experimentation and speculation. @talldan has explained that well in the above comment. 

In the meantime, I've added a comment to explain the arbitrary timeout: c7fed44b660af7f940461ad50cc9f3886a7c9ded

> The two ideas I had:

Both of those seem pretty complex to me, and I'm not sure it's worth investing in an additional complex parsing mechanism or a complex loading tracking mechanism, given that we're aiming to start async loading blocks at some point in the near future (with the upcoming work on ES modules and import maps adoption in the WordPress core).

Reply:
Agree with all your points @tyxla 👍 

Great job on getting this merged.